### PR TITLE
Deserialize bitmaps

### PR DESCRIPTION
**Pull request by mailravi3390**

### CODE REVIEW

File: core/src/main/java/net/opentsdb/aura/metrics/meta/DefaultMetaTimeSeriesQueryResult.java [hunk 1]
```diff
@@ -136,10 +140,24 @@ public void addHash(final long hash) {
...
 
+    public void addBitMap(long hash, byte[] bitMap) throws IOException {
+      //Will leave the hash as is, even though we are not using it.
+      RoaringBitmap roaringBitmap = new RoaringBitmap();
+      roaringBitmap.deserialize(ByteBuffer.wrap(bitMap));
+      bitmaps[hashCount - 1] = roaringBitmap;
+    }
+
+    public void addBitMap(long hash, RoaringBitmap bitmap) {
+      bitmaps[hashCount] = bitmap;
```
Comment (smrutilal2) [inline core/src/main/java/net/opentsdb/aura/metrics/meta/DefaultMetaTimeSeriesQueryResult.java:158]: Shouldn't this be `hashcount - 1`

File: meta-grpc-client/src/main/java/net/opentsdb/aura/metrics/meta/grpc/protobuf/BitmapGroupResultParser.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                        if (!input.skipField(tag)) {
+                            done = true;
+                        }
+                        break;
+                    }
+                }
+            }
+            return groupResult;
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (smrutilal2) [inline meta-grpc-client/src/main/java/net/opentsdb/aura/metrics/meta/grpc/protobuf/BitmapGroupResultParser.java:51]: remove the `printStackTrace()`

File: meta-grpc-client/src/main/java/net/opentsdb/aura/metrics/meta/grpc/protobuf/TimeseriesLite.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package net.opentsdb.aura.metrics.meta.grpc.protobuf;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.CodedOutputStream;
+import com.google.protobuf.MessageLite;
+import com.google.protobuf.Parser;
+
+import java.io.OutputStream;
+
+public class TimeseriesLite implements MessageLite {
```
Comment (smrutilal2) [inline meta-grpc-client/src/main/java/net/opentsdb/aura/metrics/meta/grpc/protobuf/TimeseriesLite.java:10]: This is exactly same as `BitmapGroupResultLite `. We can pull it up to a base class and reuse it.

Comment (smrutilal2) [reply meta-grpc-client/src/main/java/net/opentsdb/aura/metrics/meta/grpc/protobuf/TimeseriesLite.java:10]: ohh... `BitmapGroupResultLite ` is extending another base class. In that case we can create another Interface extending `MessageLite `  and add these as `default` methods and use that. The below code snippet would make it clear.

```
public interface DefaultMessageLite extends MessageLite {

  @Override
  default void writeTo(CodedOutputStream output){
    throw new UnsupportedOperationException();
  }
  ...
}

public class TimeseriesLite implements DefaultMessageLite {
  ...
}

public class BitmapGroupResultLite extends DefaultMetaTimeSeriesQueryResult.DefaultGroupResult implements DefaultMessageLite {
  ...
}
  
```

Comment (mailravi3390) [reply meta-grpc-client/src/main/java/net/opentsdb/aura/metrics/meta/grpc/protobuf/TimeseriesLite.java:10]: Yeah was thinking of doing that.

Comment (smrutilal2) [reply meta-grpc-client/src/main/java/net/opentsdb/aura/metrics/meta/grpc/protobuf/TimeseriesLite.java:10]: Food for thought.

Do we even need this empty classes like`DictionaryLite`, `TimeseriesLite` etc. This is definitely a code smell but we can live with it for now.

File: meta-grpc-client/src/test/java/net/opentsdb/aura/metrics/meta/grpc/TestUtil.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        for(int i = 1 ; i < count; i++) {
+            array_count[i] = array_count[i-1] - 3600 * 6;
+        }
+
+        return array_count;
+    }
+
+    public static void verifyBitMap(RoaringBitmap bitMap, int[] ts) {
+        final Iterator<Integer> iterator = bitMap.iterator();
```
Comment (smrutilal2) [inline meta-grpc-client/src/test/java/net/opentsdb/aura/metrics/meta/grpc/TestUtil.java:27]: `iterator` not used. should remove it.

File: meta-grpc-client/src/test/java/net/opentsdb/aura/metrics/meta/grpc/TestUtil.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        int i = 0;
+        for (; i < ts.length; i++) {
+            assertTrue(bitMap.contains(ts[i]));
```
Comment (smrutilal2) [inline meta-grpc-client/src/test/java/net/opentsdb/aura/metrics/meta/grpc/TestUtil.java:30]: Is the bitmap not ordered? Do we not care about the order?

Comment (mailravi3390) [reply meta-grpc-client/src/test/java/net/opentsdb/aura/metrics/meta/grpc/TestUtil.java:30]: Yeah seems like it is not ordered. We dont care about the order.

**Review by smrutilal2 — APPROVED** [review]

**Review by smrutilal2 — APPROVED** [review]
👍

**Pull request merged**